pkgs/by-name/tr/tracexec/package.nix

```diff
@@ -0,0 +1,33 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "tracexec";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "kxxt";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-HQTNYReJkfiqS05APtLAUf9dnnPYTzF0ratd5QKv+8w=";
+  };
+
+  checkFlags = [
+    "--skip=cli::test::log_mode_without_args_works" # `Permission denied` (maybe needs root?)
```

Comment:
This test doesn't need root and it only executes `/proc/self/exe` so I think it should be compatible with NixOS. Does the build environment drops some privileges, like  `CAP_SYS_PTRACE`?

Reply:
I would be surprised if `CAP_SYS_PTRACE` was available in the build sandbox.

Because `CAP_SYS_PTRACE` allows tracing arbitrary processes; I expect that tracing your own children would be allowed.

Reply:
>  I expect that tracing your own children would be allowed.

Ah, I didn't read too much about `CAP_SYS_PTRACE`.

>  I expect that tracing your own children would be allowed.


Yes, that's enough for this test.